packages/react/src/components/FileUploader/FileUploaderDropContainer.js

```diff
@@ -0,0 +1,173 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { settings } from 'carbon-components';
+import { keys, matches } from '../../internal/keyboard';
+import uniqueId from '../../tools/uniqueId';
+
+const { prefix } = settings;
+
+export default function FileUploaderDropContainer(props) {
+  const inputRef = useRef();
+  const {
+    accept,
+    className,
+    id,
+    disabled,
+    labelText,
+    multiple,
+    name,
+    onAddFiles,
+    role,
+    tabIndex,
+    ...other
+  } = props;
+  const uid = useRef(uniqueId());
+  const [isActive, setActive] = useState(false);
+  const labelClasses = classNames(`${prefix}--file-browse-btn`, {
+    [`${prefix}--file-browse-btn--disabled`]: disabled,
+  });
+  const dropareaClasses = classNames(`${prefix}--file__drop-container`, {
+    [`${prefix}--file__drop-container--drag-over`]: isActive,
+    [className]: className,
+  });
+  const handleChange = evt =>
+    onAddFiles(evt, {
+      addedFiles:
+        evt.type === 'drop'
+          ? [...evt.dataTransfer.files]
+          : [...evt.target.files],
+    });
+
+  return (
+    <div
+      className={`${prefix}--file`}
+      onDragOver={evt => {
+        evt.stopPropagation();
+        evt.preventDefault();
+        if (disabled) {
+          return;
+        }
+        setActive(true);
+        evt.dataTransfer.dropEffect = 'copy';
+      }}
+      onDragLeave={evt => {
+        evt.stopPropagation();
+        evt.preventDefault();
+        if (disabled) {
+          return;
+        }
+        setActive(false);
+        evt.dataTransfer.dropEffect = 'move';
+      }}
+      onDrop={evt => {
+        evt.stopPropagation();
+        evt.preventDefault();
+        if (disabled) {
+          return;
+        }
+        setActive(false);
+        handleChange(evt);
+      }}>
+      <label
+        className={labelClasses}
+        htmlFor={id || uid.current}
+        role={role || 'button'}
+        tabIndex={tabIndex || 0}
+        onKeyDown={evt => {
+          if (matches(evt, [keys.Enter, keys.Space])) {
+            inputRef.current.click();
+          }
+        }}
+        {...other}>
+        <div className={dropareaClasses}>
+          {labelText}
+          <input
+            type="file"
+            id={id || uid.current}
+            className={`${prefix}--file-input`}
+            ref={inputRef}
+            tabIndex="-1"
+            disabled={disabled}
+            accept={accept}
```

Comment:
Did you test this? I was able to include a .mp4 file without any issues.

Reply:
added file extension and mime type checks for drop uploads